From w.sang@pengutronix.de  Thu Apr 22 16:39:35 2010
From: Wolfram Sang <w.sang@pengutronix.de>
Date: Sat, 20 Mar 2010 15:13:02 +0100
Subject: Staging: iio: adc: fix dangling pointers
To: kernel-janitors@vger.kernel.org
Cc: devel@driverdev.osuosl.org, Wolfram Sang <w.sang@pengutronix.de>, Greg Kroah-Hartman <gregkh@suse.de>, linux-i2c@vger.kernel.org, linux-kernel@vger.kernel.org
Message-ID: <1269094385-16114-22-git-send-email-w.sang@pengutronix.de>


Fix I2C-drivers which missed setting clientdata to NULL before freeing the
structure it points to. Also fix drivers which do this _after_ the structure
was freed already.

Signed-off-by: Wolfram Sang <w.sang@pengutronix.de>
Acked-by: Jonathan Cameron <jic23@cam.ac.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/iio/adc/max1363_core.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/staging/iio/adc/max1363_core.c
+++ b/drivers/staging/iio/adc/max1363_core.c
@@ -557,6 +557,7 @@ error_put_reg:
 	if (!IS_ERR(st->reg))
 		regulator_put(st->reg);
 error_free_st:
+	i2c_set_clientdata(client, NULL);
 	kfree(st);
 
 error_ret:
@@ -574,6 +575,7 @@ static int max1363_remove(struct i2c_cli
 		regulator_disable(st->reg);
 		regulator_put(st->reg);
 	}
+	i2c_set_clientdata(client, NULL);
 	kfree(st);
 
 	return 0;
